person.rb

```diff
@@ -1,8 +1,17 @@
-class Person
+# Define the Nameable class with the correct_name method that raises a NotImplementedError
+class Nameable
+  def correct_name
+    raise NotImplementedError, 'Subclasses must implement the correct_name method.'
+  end
+end
+
```

Comment:
- [ ] Now your Decorator class in your `decorator.rb` is inherited from the Nameable class :+1:  But your Nemeable class is not required in the `decorator.rb` file. I suggest that you create your Nameable class in a separate file and require the class in this `person.rb` and in your `decorator.rb` to avoid running issues. :blush: